### PR TITLE
Comments fixes

### PR DESCRIPTION
**Pull request by hira-iqbal**

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -54,6 +53,6 @@ def set_article
   end
 
   def article_params
-    params.require(:article).permit(:title, :body, :status, images:[])
+    params.require(:article).permit( :title, :body, :status, images:[] )
```
Comment (aliammaar) [inline app/controllers/articles_controller.rb]: why these spaces?

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -46,6 +46,6 @@ def set_user
   end
 
   def user_params
-    params.require(:user).permit(:name, :surname, :email, :password, :role_id, :image)
+    params.require(:user).permit( :name, :surname, :email, :password, :role_id, :image )
```
Comment (aliammaar) [inline app/controllers/users_controller.rb]: same as above.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -1,36 +1,38 @@
 <!DOCTYPE html>
 <html>
-  <head>
```
Comment (aliammaar) [inline app/views/layouts/application.html.erb]: the new indentations is not correct. It was fine in earlier version.

File: app/views/users/_form.html.erb [hunk 4]
```diff
@@ -10,26 +10,28 @@
...
     <div class= "form-group col-md-4">
       <%= form.label :password %>
-      <%= form.password_field :password, class: "form-control", placeholder: "Password"%><br><br>
+      <%= form.password_field :password, class: "form-control", placeholder: "Password" %><br><br>
     </div>
   </div>
 
+  <div class="form-group col-md-4">
+    <% if user.image.attached? %>
+    <%= image_tag user.image, style: "width:200px", height:"100px" %>
```
Comment (aliammaar) [inline app/views/users/_form.html.erb]: indentation issue.

**Review by aliammaar — CHANGES_REQUESTED** [review]

File: app/controllers/articles_controller.rb (hunk 1 above)
Comment (hira-iqbal) [reply app/controllers/articles_controller.rb]: Spaces have been removed.

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (hira-iqbal) [reply app/controllers/users_controller.rb]: Removed

File: app/views/layouts/application.html.erb (hunk 3 above)
Comment (hira-iqbal) [reply app/views/layouts/application.html.erb]: Indentation has been fixed.

File: app/views/users/_form.html.erb (hunk 4 above)
Comment (hira-iqbal) [reply app/views/users/_form.html.erb]: Fixed

File: app/controllers/articles_controller.rb [hunk 5]
```diff
@@ -17,43 +17,42 @@ def new
   def edit; end
 
   def create
-    @article = Article.new(article_params)
+    @article = current_user.articles.new(article_params)
+    authorize @article
```
Comment (aliammaar) [inline app/controllers/articles_controller.rb:21]: `authorize @article` in `set_article` not here 🙏

File: app/controllers/articles_controller.rb [hunk 6]
```diff
@@ -17,43 +17,42 @@ def new
...
   end
+end
 
-  def destroy
-    @article.destroy
-    redirect_to articles_path
+def update
+  if @article.update(article_params)
+    redirect_to @article
     authorize @article
```
Comment (aliammaar) [inline app/controllers/articles_controller.rb]: same as above.

File: app/controllers/articles_controller.rb [hunk 7]
```diff
@@ -17,43 +17,42 @@ def new
...
+  else
+    render :edit, alert: @article.errors.full_messages
   end
+end
 
-  private
+def destroy
+  @article.destroy
+  redirect_to articles_path
+  authorize @article
```
Comment (aliammaar) [inline app/controllers/articles_controller.rb]: same as above.

File: app/views/articles/index.html.erb [hunk 8]
```diff
@@ -13,10 +13,10 @@
         <td><%= link_to article.title, article %></td>
         <td><%= link_to article.body, article %></td>
         <% article.images.each do |image|%>
-          <td><%= image_tag(article.images[image]) %></td>
+          <td><%= image_tag image, height:"100px", width:"100px" %></td>
```
Comment (aliammaar) [inline app/views/articles/index.html.erb]: Please always use `()` for a function. i.e `image_tag(image, height:"100px", width:"100px")`

File: app/views/articles/show.html.erb [hunk 9]
```diff
@@ -5,27 +5,29 @@
...
       <td><%= @article.title %></td>
       <td><%= @article.body %></td>
       <td>
-      <% @article.images.each do |image| %>
+        <% @article.images.each do |image| %>
         <%= image_tag image, height: "200px", width: "200px" %>
-       <% end %>
+        <% end %>
       </td>
+      <td><%= @article.user.name %></td>
```
Comment (aliammaar) [inline app/views/articles/show.html.erb:16]: Did we eager load `user` ?

File: app/views/comments/_comment.html.erb [hunk 10]
```diff
@@ -3,18 +3,27 @@
...
+      <th scope="col">Image</th>
+      <th scope="col">commenter Name</th>
     </tr>
   </thead>
   <tbody>
     <tr>
       <td><%= comment.commenter %></td>
       <td><%= comment.body %></td>
-      <td align="center">
+      <td>
```
Comment (aliammaar) [inline app/views/comments/_comment.html.erb]: this part being duplicated many times. Do you think we should move it to a partial?

File: app/views/users/_form.html.erb [hunk 11]
```diff
@@ -10,26 +10,27 @@
...
-
     <div class= "form-group col-md-4">
       <%= form.label :password %>
-      <%= form.password_field :password, class: "form-control", placeholder: "Password"%><br><br>
+      <%= form.password_field :password, class: "form-control", placeholder: "Password" %><br><br>
     </div>
   </div>
 
+  <div class="form-group col-md-4">
+    <% if user.image.attached? %>
```
Comment (aliammaar) [inline app/views/users/_form.html.erb]: Can we make it one liner? `<%= image_tag user.image, style: "width:200px", height:"100px" if user.image.attached? %>`

File: app/views/users/index.html.erb [hunk 12]
```diff
@@ -7,12 +7,12 @@
   </thead>
   <tbody>
 
-    <% @users.each do |user|%>
+    <% @users.each do |user| %>
       <tr>
-        <td><%= link_to user.name, user%></td>
+        <td><%= link_to user.name, user %></td>
```
Comment (aliammaar) [inline app/views/users/index.html.erb]: Use `()` please

**Review by aliammaar — CHANGES_REQUESTED** [review]

File: app/views/comments/_comment.html.erb (hunk 10 above)
Comment (hira-iqbal) [reply app/views/comments/_comment.html.erb]: updated

File: app/views/articles/show.html.erb (hunk 9 above)
Comment (hira-iqbal) [reply app/views/articles/show.html.erb:16]: Updated

File: app/views/articles/index.html.erb (hunk 8 above)
Comment (hira-iqbal) [reply app/views/articles/index.html.erb]: Updated

File: app/controllers/articles_controller.rb (hunk 6 above)
Comment (hira-iqbal) [reply app/controllers/articles_controller.rb]: Updated

File: app/controllers/articles_controller.rb (hunk 7 above)
Comment (hira-iqbal) [reply app/controllers/articles_controller.rb]: Updated

File: app/views/users/_form.html.erb (hunk 11 above)
Comment (hira-iqbal) [reply app/views/users/_form.html.erb]: Done

File: app/views/articles/_display_images.html.erb [hunk 13]
```diff
@@ -0,0 +1,5 @@
+<% images.each do |image| %>
+  <%= image_tag image, height: "200px", width: "200px" %>
+<% end %>
+
```
Comment (aliammaar) [inline app/views/articles/_display_images.html.erb]: why these extra lines.

File: app/views/comments/_comment.html.erb [hunk 14]
```diff
@@ -3,18 +3,23 @@
...
     </tr>
   </thead>
   <tbody>
     <tr>
       <td><%= comment.commenter %></td>
       <td><%= comment.body %></td>
-      <td align="center">
+      <td><%= render 'display_images', images: comment.images %></td>
+      <td><%= comment.user.name%></td>
+
```
Comment (aliammaar) [inline app/views/comments/_comment.html.erb]: why this new line.

File: app/views/users/index.html.erb (hunk 12 above)
Comment (hira-iqbal) [reply app/views/users/index.html.erb]: Done

**Review by aliammaar — APPROVED** [review]

**Pull request merged**